pkg/deployer/baremetal/artifacts.go

```diff
@@ -200,16 +206,33 @@ func (am *ArtifactManager) download(typ ArtifactType, version, pkgDir string) (s
 	}
 
 	httpClient := &http.Client{}
-	req, err := http.NewRequest(http.MethodGet, downloadURL, nil)
-	if err != nil {
-		return "", err
-	}
 
 	am.logger.V(3).Infof("Downloading artifact from '%s' to '%s'", downloadURL, artifactFile)
 
-	resp, err := httpClient.Do(req)
+	resp, err := am.startDownload(downloadURL, httpClient)
 	if err != nil {
-		return "", err
+		downloadURL, err = am.artifactURL(typ, version, TarGzExtension)
```

Comment:
Is this code duplicated with the code above?

```go
if !am.alwaysDownload {
     // ...
}
```

Maybe we don't need `startDownload()` and just handle the HTTP status code.

Reply:
This is related to the above discussion. If the suffix name is determined first, it can avoid two HTTP requests.